core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -300,4 +302,12 @@ protected static void throwParsingExceptionOnMultipleFields(String queryName, XC
                     + processedFieldName + "] and [" + currentFieldName + "]");
         }
     }
+
+    /**
+     * Adds 'boost' and 'query_name' parsing to all query builder parsers passed in
+     */
+    protected static void declareStandardFields(AbstractObjectParser<? extends QueryBuilder, ? extends ParseFieldMatcherSupplier> parser) {
+        parser.declareFloat((builder, value) -> builder.boost(value), AbstractQueryBuilder.BOOST_FIELD);
```

Comment:
You mean `AbstractQueryBuilder::boost` I tried this for a while now but can't get it to work, there's something going on with the generics in AbstractQueryBuilder that seems to prevent this. I'll leave it for now as it seems to work.